CONTRIBUTING.md

```diff
@@ -0,0 +1,46 @@
+# Contributing to parity-common
+
+parity-common welcomes contribution from everyone in the form of suggestions, bug
+reports, pull requests, and feedback. This document gives some guidance if you
+are thinking of helping us.
+
+Please reach out here in a GitHub issue if we can do anything to help you contribute.
+
+## Submitting bug reports and feature requests
+
+When reporting a bug or asking for help, please include enough details so that
+the people helping you can reproduce the behavior you are seeing. For some tips
+on how to approach this, read about how to produce a [Minimal, Complete, and
+Verifiable example].
+
+[Minimal, Complete, and Verifiable example]: https://stackoverflow.com/help/mcve
+
+When making a feature request, please make it clear what problem you intend to
+solve with the feature, any ideas for how parity-common could support solving that problem, any possible alternatives, and any disadvantages.
+
+## Versioning
+
+As many crates in the rust ecosystem, all crates in parity-common follow [semantic versioning]. This means bumping PATCH version on bug fixes that don't break backwards compatibility, MINOR version on new features and MAJOR version otherwise (MAJOR.MINOR.PATCH). Versions < 1.0 are considered to have the format 0.MAJOR.MINOR, which means bumping MINOR version for all non-breaking changes.
+
+If you bump a dependency that publicly exposed in crate's API (e.g. `pub use dependency;` or `pub field: dependency::Dependency`) and the version transition for dependency was semver-breaking, than it is considered to be a breaking change. To put it simply, if you change could cause a compilation error in user's code, it is a breaking change.
+
+## Releasing a new version
+
+When making a new release make sure to follow these steps:
```

Comment:
Let's make it clear our preference here: should PRs include version bump and tag or is this up to us to handle?

So fare we've been a bit hazy on this. I tend to include version bumps along with the code changes, but I don't think that's what we want external contributors to do  (i.e. not part of the `paritytech` group of cargo maintainers).
Also, let's clarify that the tagging should happen after the squashed PR commit is on master (and show them the command, lest people forget to push the tag).